snuba/request/schema.py

```diff
@@ -62,15 +68,17 @@ def __init__(
 
     @classmethod
     def build_with_extensions(
-        cls, extensions: Mapping[str, QueryExtension]
+        cls,
+        extensions: Mapping[str, QueryExtension],
+        settings_class: Type[RequestSettings] = HTTPRequestSettings,
```

Comment:
`build_with_extensions` only has two call sites on the HTTP request path, so I think it might be better to make this non-optional?